viewing_party/party.py

```diff
@@ -1,23 +1,156 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    if title is None or genre is None or rating is None:
+        return None
+    return {
+        "title": title,
+        "genre": genre,
+        "rating": rating
+    }
+
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+
+def watch_movie(user_data, movie_title):
+    for movie in user_data["watchlist"]:
+        if movie["title"] == movie_title:
+            user_data["watchlist"].remove(movie)
+            user_data["watched"].append(movie)
+    return user_data
+
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+def get_watched_avg_rating(user_data):
+    avg_rating = 0
+
+    if len(user_data["watched"]) > 0:

```

Comment:
Ahh I thought saving on lines would be preferred since it's checking if the list is populated and if not, it jumps down to the return. Will try and get in the habit of guard clause for edge cases first, thanks!